laserscan_kinect/include/laserscan_kinect/laserscan_kinect.h

```diff
@@ -206,17 +210,35 @@ class LaserScanKinect
                 {
                     if (depth_m < depth_min && depth_raw_mm < dist_to_ground_corrected[i])
                     {
-                        depth_min = depth_m;
+                        // Check the detected object is a slope
+                        if ((depth_prev - depth_prev_prev) < min_diff && (depth_m - depth_prev) < min_diff)
```

Comment:
would be nice to have a class variable that decides whether we want to detect ramps and/or the angle at which we want to consider something to be a ramp. Then we can assign the value of `min_diff` based on this and only execute the new blocks that you have introduced when `ramp_detection_ = true`. This boolean variable can be passed in from a launch file, making it easy to enable/disable the feature, and more likely to be accepted as a change upstream (cause these changes will likely break/affect functionality for people already using this package for other things)

Reply:
Yes that makes a lot of sense to do, I think it would be good to be able to configure min_diff, etc. in the config file. The only issue with specifying an angle is that it might vary with different cameras. I will look into it a bit more though